torch/_inductor/codegen/wrapper.py

```diff
@@ -147,10 +147,26 @@ class EnterCudaDeviceContextManagerLine:
     def codegen(self, code: IndentedBuffer, device_cm_stack: contextlib.ExitStack):
         if V.graph.cpp_wrapper:
             code.writeline("\n")
-            if self.first_time:
-                code.writeline(f"at::cuda::CUDAGuard device_guard({self.device_idx});")
+            if V.graph.aot_mode:
+                # In AOT mode, we have a stream provided as a param. A stream is
+                # associated with a device, so it doesn't make sense for the device
+                # to change.
+                assert self.first_time
+                # CUDAStreamGuard sets the stream and the device.
+                code.writeline(
+                    f"at::cuda::CUDAStreamGuard stream_guard("
+                    f"at::cuda::getStreamFromExternal(stream, {self.device_idx}));"
+                )
             else:
-                code.writeline(f"device_guard.set_index({self.device_idx});")
+                if self.first_time:
+                    code.writeline(
+                        f"at::cuda::CUDAGuard device_guard({self.device_idx});"
+                    )
+                else:
+                    code.writeline(f"device_guard.set_index({self.device_idx});")
+                code.writeline(
+                    f"stream = at::cuda::getCurrentCUDAStream({self.device_idx});"
```

Comment:
You can follow how stream is get and passed in the python wrapper codegen, https://github.com/pytorch/pytorch/blob/a361fceef39766e839fbfe813d195339c66b5391/torch/_inductor/codegen/wrapper.py#L717C1-L717C1